docs/internals/layout_in_storage.rst

```diff
@@ -53,30 +64,56 @@ Mappings and Dynamic Arrays
 
 .. _storage-hashed-encoding:
 
-Due to their unpredictable size, mapping and dynamically-sized array types use a Keccak-256 hash
-computation to find the starting position of the value or the array data.
-These starting positions are always full stack slots.
+Due to their unpredictable size, mappings and dynamically-sized array types cannot be stored
```

Comment:
Hm, not sure about this. If you don't know the size of something but can put a reasonable upper bound on it, you might still store it in-place.

Reply:
In the case of mappings and dynamically-sized arrays isn't the reason they are not stored in between because their size is unknown (because this value set at runtime)? I agree that in the hypothetical case where some data structure is bounded but exact size unknown, one could still store "in between" but that isn't the case here iiuc

Reply:
What about "Since their size is not fixed, mappings..."?